### PR TITLE
[SPARK-32158] [SQL] Add options to toJSON function

### PR DESCRIPTION
**Pull request by Gschiavon**
Now there is a `toJSON` function and I added `toJSON(options)` . I duplicated code now but the idea would be to have `toJSON()` and `toJSON(options)` so we could reuse all the code. The only problem is that adding the `()` to `toJSON` is a breaking change and I want to see what do you think about this.

For reference see  https://issues.apache.org/jira/browse/SPARK-23772.
### What changes were proposed in this pull request?
Adding JSON Option for `toJSON` 

### Why are the changes needed?
In order to be consistent with `to_json` and to be able to print null values when calling `toJSON`

### Does this PR introduce _any_ user-facing change?
Depends, the idea would be to use the original `toJSON` but the function doesn`t accept parameters, and if I add them, could be a break change.

### How was this patch tested?
Just added a test to show the use case. The idea is to reuse all the existing test cases plus some with null values.

Jira issue -> https://issues.apache.org/jira/browse/SPARK-32158

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -3419,6 +3419,39 @@ class Dataset[T] private[sql](
     } (Encoders.STRING)
   }
 
+  /**
+   * Returns the content of the Dataset as a Dataset of JSON strings.
+   * @since 3.0.1
+   */
+  def toJSON(options: Map[String, String]): Dataset[String] = {
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3426]: The built-in `to_json` is not enough for your usecase? cc: @HyukjinKwon

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 2]
```diff
@@ -3419,6 +3419,39 @@ class Dataset[T] private[sql](
     } (Encoders.STRING)
   }
 
+  /**
+   * Returns the content of the Dataset as a Dataset of JSON strings.
+   * @since 3.0.1
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3424]: note: we cannot add a new feature in v3.0.x, so not `3.0.1` but `3.1.0`.

Comment (Gschiavon) [reply sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3424]: ok!

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala (hunk 1 above)
Comment (Gschiavon) [reply sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3426]: I guess so, you'd have to do something like this `df.select(to_json(struct("*"), Map("ignoreNullFields" -> "false")).as("value"))` to produce the same result as `df.toJSON(options)`

Comment (maropu) [reply sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3426]: Yea, I think so.

Comment (HyukjinKwon) [reply sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3426]: I don't feel strongly here. I would rather don't extend `toJSON` anymore given that we added `to_json`, `from_json`, etc. to cover such general cases. I am okay if any committer supports it.

Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3426]: I would deduplicate it by calling `toJSON(options: Map[String, String])` at `toJSON`.

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 3]
```diff
@@ -3441,6 +3474,7 @@ class Dataset[T] private[sql](
     files.toSet.toArray
   }
 
+
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3477]: Let's remove unrelated newline changes.

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala (hunk 1 above)
Comment (Gschiavon) [reply sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3426]: ok!

Comment (maropu) [reply sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3405]: Thanks for the check, @HyukjinKwon. I'm neutral on this, too.

**Comment by Gschiavon** [comment]
what should we do @HyukjinKwon ? I think it is simplier this way but I don't have a strong opinion either, as there are workarounds.

**Comment by HyukjinKwon** [comment]
You can wait for other committers to review. Otherwise, I would just close.
